### PR TITLE
Knausj generic terminal establishment

### PR DESCRIPTION
**Pull request by DonnerWolfBach**
solves #505

### CODE REVIEW

File: apps/platforms/win/powershell/power_shell.py [hunk 1]
```diff
@@ -97,6 +94,10 @@ def terminal_clear_screen():
     def terminal_run_last():
         actions.key("up enter")
 
+    def terminal_rerun_search(command: str):
+        actions.key("ctrl-r")
```
Comment (fidgetingbits) [inline apps/platforms/win/powershell/power_shell.py:98]: TIL powershell supports this lol

**Review by fidgetingbits — APPROVED** [review]
looks good to me

**Review by knausj85 — APPROVED** [review]

**Comment by knausj85** [comment]
Looks great, once you think it's ready we will merge it

**Comment by DonnerWolfBach** [comment]
I think there are some things left to do, like deciding wether terminal_list_all & co should be defined in the terminal specific file or in generic_terminal.py.

One question about the existing terminals: should they all be first transitioned to the new "architecture" or should this be done in later PRs?

File: apps/platforms/win/powershell/power_shell.py (hunk 1 above)
Comment (DonnerWolfBach) [reply apps/platforms/win/powershell/power_shell.py:98]: To be honest, me too.

**Comment by knausj85** [comment]
Generally, things should work -- so the existing terminals should be migrated where it's reasonable. This does introduce some commands that will not work for e.g. gnome-terminal and guake as-is. I personally can live with this though.

I'm firmly on the fence re: "terminal_list_all & co should be defined in the terminal specific". I'm fine with the current approach, but we could have a `generic_unix_shell` tag for these super common implementation for terminals to opt into, as fidget suggested.

**Pull request merged**